httpserver/bin/note_server.dart

```diff
@@ -6,67 +6,85 @@
 // Use note_taker.html to run the client.
 
 import 'dart:io';
-import 'dart:convert' show JSON;
+import 'dart:async';
+import 'dart:convert' show UTF8, JSON;
 
 int count = 0;
 
-void main() {
+main() async {
   // One note per line.
-  List<String> lines = new File('notes.txt').readAsLinesSync();
-  count = lines.length;
+  try {
+    List<String> lines = new File('notes.txt').readAsLinesSync();
+    count = lines.length;
 
-  HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4042)
-      .then(listenForRequests)
-      .catchError((e) => print('hello: ${e.toString()}'));
+    var server =
+        await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4042);
+    print('Listening for requests on 4042.');
+    await listenForRequests(server);
+  } on FileSystemException {
+    print('Could not open notes.txt');
```

Comment:
Split the try into two parts - one for the reading  of notes.txt and one for the loop. It is confusing to have the FileSystemException catch here, after the listen-loop, because it's not at all celar that the listen loop can't throw such an exception as well.

Actually, I don't think catching errors and printing "hello: $e" is a good example to set, so just leave out the error handling and let it be uncaught.

So:
```dart
var lines;
try {
   lines = new FIle('notes.txt').readAsLinesSync();
} on FileSystemException {
  print('Could not open notes.txt');
  return;
}
var server = await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4042);
print('Listening for requests on 4042.');
await listenForRequests(server);